cmd/cayley/command/repl.go

```diff
@@ -101,9 +102,19 @@ func NewQueryCmd() *cobra.Command {
 		Aliases: []string{"qu"},
 		Short:   "Run a query in a specified database and print results.",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if len(args) != 1 {
-				return fmt.Errorf("expected query as an argument")
+			var querystr string
+			if len(args) == 0 {
+				bytes, err := ioutil.ReadAll(os.Stdin)
+				if err != nil {
+					panic(err)
```

Comment:
As I mentioned earlier, you should just return an error here instead of throwing panic.